compiler/rustc_infer/src/infer/mod.rs

```diff
@@ -973,9 +973,6 @@ impl<'tcx> InferCtxt<'tcx> {
             TypingMode::Analysis { defining_opaque_types } => {
                 id.into().as_local().is_some_and(|def_id| defining_opaque_types.contains(&def_id))
             }
-            // FIXME(#132279): This function is quite weird in post-analysis
-            // and post-borrowck analysis mode. We may need to modify its uses
-            // to support PostBorrowckAnalysis in the old solver as well.
```

Comment:
why remove this FIXME :thinking: I feel like ideally we just never call this function outside of `TypingMode::Analysis`, not sure whether that's possible, might be kinda ugly to implement